### PR TITLE
cmd/snap: use ⬏ instead of ↑ where applicable

### PR DESCRIPTION
**Pull request by anonymouse64**
This makes the output more clear that a given channel is actually tracking the
one above it. See:
```
  1/stable:    2.10 2018-12-18   (1) 65kB -
  1/candidate: ↑
```
vs the more clear
```
  1/stable:    2.10 2018-12-18   (1) 65kB -
  1/candidate: ⬏
```
Probably snapcraft should eventually do the same thing.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Nice, I like this

**Comment by anonymouse64** [comment]
Yeah I missed that we also have spread tests around this, that is fixed now too

**Comment by diddledan** [comment]
Copied from my comment in the forum for historical awareness about the decision: I think that ↑ was chosen because it is within the base character set while ⬏ requires an extended character set.

**Comment by anonymouse64** [comment]
> I think that ↑ was chosen because it is within the base character set while ⬏ requires an extended character set.

IMHO, using ⬏ presents a better interface than ↑, and if we have reports of regressions in usability from folks with ⬏ then we can go back (this PR is quite simple and thus quite easy to revert), but I think the majority of terminals out there will support ⬏ just fine.

**Comment by diddledan** [comment]
Cool, yeah, I've just read through the code now and see that the fancy character is only used when the terminal is Unicode capable. So ASCII-only boxes still see the `^` symbol and fancypants terminals see the new shiny :+1: .

File: cmd/snap/color.go [hunk 1]
```diff
@@ -37,11 +37,11 @@ type unicodeMixin struct {
 func (ux unicodeMixin) addUnicodeChars(esc *escapes) {
 	if canUnicode(ux.Unicode) {
 		esc.dash = "–" // that's an en dash (so yaml is happy)
-		esc.uparrow = "↑"
+		esc.rightUpArrow = "⬏"
 		esc.tick = "✓"
 	} else {
 		esc.dash = "--" // two dashes keeps yaml happy also
-		esc.uparrow = "^"
+		esc.rightUpArrow = "^"
```
Comment (diddledan) [inline cmd/snap/color.go:44]: Seeing as we're editing this line, anyhow, would the following suggestion help to improve the understanding of its meaning at all?

```suggestion
		esc.rightUpArrow = "-^"
```

**Comment by mvo5** [comment]
> Copied from my comment in the forum for historical awareness about the decision: I think that ↑ was chosen because it is within the base character set while ⬏ requires an extended character set.

I wonder if @chipaca has an opinion here? (sorry for pining you here John)

**Comment by mvo5** [comment]
Hrm, what is interesting is that in xterm the ⬏ is indeed not displayed but the  ↑ is :/

**Comment by chipaca** [comment]
I do! I saw the PR but abstained from commenting despite having opinions, but thank you for inviting me to share my thoughts, because now I will :smile: 

`⬏` is IMHO better than `↑` as I think @diddledan pointed out, and it reads better, but it does have problems.

The first issue is that `↑` has been adopted across the board to mean what it does in `snap info`, and so the conversation should probably be broader than just here. I'm thinking of both snapcraft and charmcraft, here, and it's possible juju also (I'm not current on their progress on this front).

The second is that `↑` is part of the default terminal font (on Ubuntu at least) whereas `⬏` shows up as �¹ meaning that the mechanism to determine which glyph to show becomes more complicated than `cmd/snap/color.go`'s `canUnicode` (or we drop console out to 'no unicode' land).

(this reminds me that we're very careful with unicode in our layout and then just blat the description out, with hilarious consequences -- there's a bug filed about this somewhere)

HTH, HAND!

---
1. note that the tick also shows like this, but as the console replacement character is just a diamond, and the tick fallback is an asterisk, we decided it was fine to leave it at that,

**Comment by anonymouse64** [comment]
Well I tried...
`↑` has been okay for long enough, I don't really have the desire or bandwidth to push this through across the board

**Pull request closed without merging**